### PR TITLE
Feature#6 documentation

### LINKED ISSUE #6: Automatically document this project
# What

A full documentation should be available for both users and developers of the project.

# Why

To help people use the bot as well as developer join the project

# Who will benefit

Developers, Users

# Functional criteria

  * The documentation should be available to user through a website
  * The dev documentation should be available within the repository
  * The dev documentation should includes all available commands
  * The user documentation should be automatically updated when the bot is updated

### PR DESCRIPTION
**Pull request by Terag**
This pull request:

 - closes #6 :
   - Documenting the code for use of godoc by the developers
   - Creating docs folder for md linked to github page documenting the project. This documentation addresses: users, developers and future contributors. (The github page is automatically generated from md files in the docs folder. Will be visible on https://terag.github.io/kadok/ and http://kadok.pedestres.fr/ when the master is updated)
   - Add the credit and contributors information

### CODE REVIEW

**Comment by sarlam** [comment]
side note, the refactoring should have been done in a separate merge request, it is not relative to documentation itself

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,7 @@
 *.out
 
 #.vscode configuration files
-.vscode/*
\ No newline at end of file
+.vscode/*
+
+#Goland configuration files
+.idea/*
```
Comment (sarlam) [inline .gitignore:18]: this is not `golang configuration files` but `intellij config files`

File: .gitignore [hunk 2]
```diff
@@ -12,4 +12,7 @@
 *.out
```
Comment (sarlam) [inline .gitignore:12]: it would be nice to generate a new gitignore using gitignore.io

File: README.md [hunk 3]
```diff
@@ -1,2 +1,209 @@
...
+
+# Credits
+
+The bot is developed using the SDK DiscordGo: https://github.com/bwmarrin/discordgo
+
+The bot and characters it quotes are inspired from the universe of Kaamelott by Alexandre Astier: https://kaamelott.com/
+
+# License
+
+GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007
```
Comment (sarlam) [inline README.md:209]: we should probably license Kaamelott as well

File: .gitignore (hunk 1 above)
Comment (Terag) [reply .gitignore]: Corrected with the new gitignore file generated from gitignore.io as suggested

File: .gitignore (hunk 2 above)
Comment (Terag) [reply .gitignore:17]: Done

File: README.md (hunk 3 above)
Comment (Terag) [reply README.md]: A line licensing Kaamelott has been added. Waiting for confirmation

**Comment by Terag** [comment]
The merge request has been reworked has requested. It now includes changes regarding the Documentation only. The other modifications are moved to a new branch for a future PR.

**Comment by Terag** [comment]
As discussed with @sarlam today, we should consider removing direct references and resources concerning Kaamelott in order to make the bot more generic and easily reusable by other communities (example: Harry Potter fans).
This point may be discussed further with the integration of docker.

As of today, the credit will be left as the project still contains quotes from Kaamelott characters

>Et qu'est-ce que vous voulez que j'y fasse à la paperasse, moi ? Vous me prenez pour un serre-livre ?

>And what do you want me to do with the paperwork? Do you think I'm a bookend?

_Leodagan_

File: docs/_config.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+title: Kadok
+remote_theme: pmarsceill/just-the-docs
+description: "Where's Kadoc? Is he well hidden?"
+footer_content: "Kadok is maintained by the team of \"Les Petits Pédestres\" under the <a href=\"https://github.com/Terag/kadok/blob/master/LICENSE\">GNU GENERAL PUBLIC LICENSE Version 3.</a>"
+color_scheme: nil
```
Comment (sarlam) [inline docs/_config.yml:5]: why is there no `color_scheme` ? is it mandatory to specify it as `nil` ?

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Contributing to Kadok's development
+
+Hi there! We are happy that you would like to contribute to Kadok!
+
+Kadok is an open source discord bot. Following these development guidelines will help to keep the project clean and running. 
+
+## How to report a bug
```
Comment (sarlam) [inline docs/contributing.md:7]: side note: we should probably create templates to report bug or ask for new features

File: docs/_config.yml (hunk 4 above)
Comment (Terag) [reply docs/_config.yml:5]: From the theme [documentation](https://github.com/pmarsceill/just-the-docs/blob/master/_config.yml):
> Color scheme currently only supports "dark" or nil (default)

It is not very clear at first, a comment could be added for future modification.

**Pull request merged**